accio-main/src/main/java/io/accio/main/wireprotocol/auth/FileAuthentication.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.accio.main.wireprotocol.auth;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Map;
+
+public class FileAuthentication
+        implements Authentication
+{
+    private static final Logger LOG = Logger.get(FileAuthentication.class);
+    private final Map<String, String> accounts;
+
+    public FileAuthentication(File authFile)
+    {
+        this.accounts = authFile.exists() && authFile.isFile() ? parseAuthFile(authFile) : null;
```

Comment:
`exists()` is redundant and `isFile()` javadoc says it true if and only if the file denoted by this abstract pathname exists and is a normal file; false otherwise

```suggestion
        this.accounts = authFile.isFile() ? parseAuthFile(authFile) : ImmutableMap.of();
```